### PR TITLE
chore: Mark Token class as immutable

### PR DESCRIPTION
**Pull request by mvorisek**
`Token` class is already fully used as immutable. `Tokens` class relies on immutable `Token`.

Marking the class as immutable helps phpstan to understand the immutability.

related with #8228 caching issue

### CODE REVIEW

File: src/Tokenizer/Token.php [hunk 1]
```diff
@@ -20,7 +20,10 @@
  * Representation of single token.
  * As a token prototype you should understand a single element generated by token_get_all.
  *
+ * @immutable
```
Comment (keradus) [inline src/Tokenizer/Token.php:23]: > Marking the class as immutable helps phpstan to understand the immutability.

can you elaborate on that?
where phpstan gets lost on immutability of Token instance? there is no mutator method in this class.
happy to see phpstan error that is present without this change and solved after this change.

File: src/Tokenizer/Token.php [hunk 2]
```diff
@@ -20,7 +20,10 @@
...
+ *
  * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ * @author Michael Vorisek <https://github.com/mvorisek>
```
Comment (keradus) [inline src/Tokenizer/Token.php:26]: only if picky - quite an addition for adding one annotation in this PR :D

File: src/Tokenizer/Token.php (hunk 1 above)
Comment (mvorisek) [reply src/Tokenizer/Token.php:23]: sure - if mutator is added (or mutation reflection/closure rebind) PHPStan will warn newly

File: src/Tokenizer/Token.php (hunk 2 above)
Comment (mvorisek) [reply src/Tokenizer/Token.php:26]: Small class, small changes :D

File: src/Tokenizer/Token.php (hunk 1 above)
Comment (keradus) [reply src/Tokenizer/Token.php:23]: thank you.
in that case, i would rather suggest to have `@readonly` annotation, applied for all DTO classes, and (as separate PR), have Fixer that converts annotation to `readonly class` syntax.

introducing `@immutable` in one place is not systematic approach

Comment (mvorisek) [reply src/Tokenizer/Token.php:23]: Do you know `@immutable` is stronger?

https://phpstan.org/r/874645f0-1413-4f4c-a5ec-ab69a5958127

> i would rather suggest to have `@readonly` annotation, applied for all DTO classes, and (as separate PR), have Fixer that converts annotation to `readonly class` syntax.

I still run almost all projects with PHP 7.4 support so coding Fixer is out of my capacity.

Comment (keradus) [reply src/Tokenizer/Token.php:23]: I do not believe it is stronger:
https://phpstan.org/r/874645f0-1413-4f4c-a5ec-ab69a5958127

please notice my previous suggestion of using `@readonly` should be only mid-step and we should convert it to `readonly class`.
(perhaps PHPStan could also treat `@readonly` as alias for @immutable`)

Comment (mvorisek) [reply src/Tokenizer/Token.php:23]: yes, see https://github.com/phpstan/phpstan/issues/11775 for explanation why this PR is perfect and `@readonly` above the class(es) is currently useless :)

Comment (mvorisek) [reply src/Tokenizer/Token.php:23]: Angly on me?

**Pull request closed without merging**